### PR TITLE
add review comment to sb files

### PR DESCRIPTION
**Pull request by oleksandr-didyk**
Contributes to https://github.com/dotnet/source-build/issues/3435

Adds comments to source-build files asking for the inclusion of the source-build team in PRs that alter `SourceBuild*` files. Non-reviewed changes could potentially cause issues down the line, be it in the downstream repos or the product build (as has happened in the past, see https://github.com/dotnet/source-build/issues/3435#issuecomment-1570650046)

### CODE REVIEW

File: eng/SourceBuild.props
Comment (JoeRobich) [inline eng/SourceBuild.props:1]: Does this work similar to a [code owners file](https://docs.github.com/en/repositories/managing-your-repositorys-settings-and-features/customizing-your-repository/about-code-owners) or will we need to manually add the reviewer?

**Review by JoeRobich — APPROVED** [review]

File: eng/SourceBuild.props
Comment (oleksandr-didyk) [reply eng/SourceBuild.props:1]: Unfortunately it does not, manual addition would be required.

Initially wanted to add both the comment and the `CODEOWNERS` entry, but for a team to be added as a code owner it needs to have [write permissions](https://docs.github.com/en/repositories/managing-your-repositorys-settings-and-features/customizing-your-repository/about-code-owners#example-of-a-codeowners-file) to the repo and I wanted to verify first if this is something we would need. I should've mentioned that in the description though, my bad, forgot to edit it.

If you are OK with granting write permissions to `dotnet/source-build-internal`, I will add the `CODEOWNERS` entry.

Comment (JoeRobich) [reply eng/SourceBuild.props:1]: I'll defer to @sharwell as I am not maintaining this repo any longer.

Comment (oleksandr-didyk) [reply eng/SourceBuild.props:1]: Reached out to Sam in DMs - agreed to add a `CODEOWNERS` entry + write access for the team tag mentioned

**Pull request merged**
